index.js

```diff
@@ -159,6 +159,12 @@ module.exports = {
       optionalDependencies: false,
     }],
 
+    // Disruptive to development and can be problematic with Redux
+    // Also, some people prefer named exports over default ones and there's
+    // nothing better or worse about each choice that should be forced
```

Comment:
>Disruptive to development

please explain how this is disruptive to development

>there's nothing better or worse about each choice

Not sure I agree, if there's only one "thing" being exported many times using that as the default export has better ergonomics for usage (I'm thinking `import logger from './logger';` vs `import { logger } from './logger';`). Please explain how preferring a default export may not improve usability, maintainability, and readability.

Reply:
There are arguments on both sides for default vs named exports. Personally, I prefer default exports, but there are lots of people who prefer named exports. There is nothing wrong with either one, and it depends entirely on context.

If the code is meant for internal use by your project, it doesn't matter which one you choose. Any refactor you make has limited impact.

If the code is meant for external use by 3rd parties, then a good argument for not using default exports and instead using named exports is that if you decide to add additional exports in the future and don't want to use a default one anymore, you won't have to release a breaking change.

As far as what I mean by disruptive to development, I mean something that impedes natural development flow.

When you first start writing code in a new file, you may not know whether or not that function will be the only export, or you might know that in the future, you definitely plan on adding more exports. Being forced by ESLint to write it as a default export now even though in the future you know you will add more exports and want them to all be named exports is disruptive to development. You will have to refactor your code when you change it but you always intended on it.

Here's just one example. Let's say you want code consistency within a certain context, such as redux reducers. It is standard practice to use named exports for reducer functions. It's possible that I might only have a single reducer for a particular part of the state right now. I shouldn't be forced to write that as a default export. It would make the import of the reducer in my unit tests look different than every other reducer import where they are named. The consistency of using named exports in this context is valid.

This rule should be disabled because there are too many valid cases when it should not be followed or would result in forcing unnecessary future refactoring.

Reply:
As far as code maintainability, readability, and usability, we all have experience using 3rd party libraries and our own code that has both named and default exports. We know how to read both, we know how to use both. One is no more or less readable than the other. It's up to the developer writing the code.

I mean, you might think that react-redux connect() should be a default export. It's almost exclusively the only thing you import from that library. Why is it a named export? Because the developer decided it should be. Does it really affect the maintainability, readability or usability of your code because it's wrapped in braces? No.

Reply:
Agree this one should be disabled. If you look through repos this rule is frequently disabled because developers tend to know when they want something to be default vs named exports. It's very common for developers to disable this rule in a file for the first export and then remove the disable comment when they add more exports.

I sort of get the idea behind this rule -- to catch someone mistakenly using a named export when they should use a default -- but I think the intended-single-named-export case is actually much more common and this rule can be a nuisance.

Reply:
I do agree with disabling. Generally speaking when bundling, named exports (eg `import { logger } from './logger';`) are easier to tree shake without extra configuration in bundlers. Practically speaking, there are times where it is desired to use a single named export (eg `export function utilOne() {}`) and the rule will error, even if there are plans to expand with other named exports in the file.